test-infra/camel-test-infra-core/src/test/java/org/apache/camel/test/infra/core/DefaultAnnotationProcessor.java

```diff
@@ -151,13 +152,12 @@ private void doInvokeFixture(Annotation annotation, Field field, Object instance
                 if (field.getType() == MockEndpoint.class) {
                     final MockEndpoint mockEndpoint = contextExtension.getMockEndpoint(uri);
 
-                    assert mockEndpoint != null;
-
+                    ObjectHelper.notNull(mockEndpoint, "mockEndpoint");
                     field.set(instance, mockEndpoint);
                 } else {
                     final Endpoint endpoint = context.getEndpoint(uri);
 
-                    assert endpoint != null;
+                    ObjectHelper.notNull(endpoint, "endpoint");
```

Comment:
A bit torn on this one, as the assert is missing a message saying what's going on.

Reply:
I think [the message here](https://github.com/apache/camel/blob/camel-4.0.0-M1/core/camel-util/src/main/java/org/apache/camel/util/ObjectHelper.java#L156) would be clear enough as:

```
endpoint must be specified
```

Reply:
Yeah, that would be ok.